api/src/wrangler.rs

```diff
@@ -157,3 +191,291 @@ impl<'b> ExecutionWrangler<'b> {
         (msg, msg_length)
     }
 }
+
+impl<'b, BS> VM for ExecutionWrangler<'b, BS>
+where
+    BS: Blockstore,
+{
+    fn blockstore(&mut self) -> &dyn Blockstore {
+        // It's unfortunate that we need to call flush here everytime we need the blockstore reference
+        // However, the state_tree inside Executor wraps whatever blockstore it's given with a BufferedBlockstore
+        // Since the store held by the Wrangler was cloned prior to being wrapped in the BufferedBlockstore, it's possible
+        // there are pending changes to the underlying blockstore held in the BufferedBlockstore's cache that are not
+        // visible via our handle
```

Comment:
DefaultMachine wraps the blockstore it's given _internally_ with a BufferedBlockstore. If it didn't do this and instead the entire stack required a BufferedBlockstore to be passed in then we could simply have the Rc around the BufferedBlockstore at the Wrangler level.